TESTING.md

```diff
@@ -0,0 +1,61 @@
+# Testing 'oneview-sdk' gem source code
+The 'oneview-sdk' source code provides a unit, integration/functional, system testing strategies, and linting tools to ensure its code style.
+This document will cover on how to execute and implement them.
+
+## Executing tests
+The tests can be executed by `rake` tasks, `guard` watches, and by their original tools, like `rspec`, and `rubocop`.
+
+### Rake
+All the test strategies and checks can be executed by the rake command.
+Please use `rake -T` to see a full list of commands.
+
+### Guard
+Guard will watch the changes and execute unit tests and style checks. Just activate it to have real time test execution on every change.
+
+### Unit tests
+All unit tests are inside the spec folder. You can execute them manually by using RSpec and specifying the test files, like `rspec spec/unit/path/to/my/tests`.
+
+### Integration tests
+The integration tests are responsible for checking if the `oneview-sdk` is correctly integrated with the OneView appliance, at the same time it tests its functionalities.
+
+#### Categories
+Each integration test has its own pre-requisites and leaves the appliance in a different post state, and given OneView composable nature, each resource needs other resources configured so it can work.
+
+The most top-level resources would need the appliance entirely configured, and setting up and tearing it down for each test would cost a lot of time (likely days). Because of the slow time to set up the tests have dependencies on each other that sped up the execution to few minutes or likely an hour depending on the test environment.
```

Comment:
> Because of the slow time to set up the tests have dependencies on each other that sped up the execution to few minutes or likely an hour depending on the test environment.

Likely rewrite this bit, its not very clear.